lms/lmsweb/views.py

```diff
@@ -236,6 +226,42 @@ def read_all_notification():
     return jsonify({'success': success_state})
 
 
+@webapp.route('/share', methods=['POST'])
+@login_required
+def share():
+    solution_id = int(request.json.get('solutionId', 0))
+    solution, shared_solution = share_link.solution_and_shared(solution_id)
+    if shared_solution is None:
+        SharedSolution.create_shared_solution(
+            solution=solution,
+        )
+        return jsonify({
+            'success': 'true',
+            'shared': True,
+        })
+
+    shared_solution.delete_instance()
```

Comment:
I believe the toggle here is a bad design for our API.

A better way might be to have a parameter to `/share/get` and `/share/delete`